code/game/objects/effects/effect_system.dm

```diff
@@ -98,6 +98,7 @@ steam.start() -- spawns the effect
 	icon = 'icons/effects/effects.dmi'
 	anchored = TRUE
 	mouse_opacity = MOUSE_OPACITY_UNCLICKABLE
+	pass_flags = PASS_FLAG_TABLE | PASS_FLAG_GRILLE
```

Comment:
`/obj/structure/fire_source` probably needs a `CanPass` override to make this work
```
/obj/structure/fire_source/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
	return ..() || (istype(mover) && mover.checkpass(PASS_FLAG_TABLE))
```
Also, should consider if we really want sparks to go through grilles or not. I think just tables is fine?